utils/dbt_format_utils.py

```diff
@@ -0,0 +1,329 @@
+    else:
+        raise Exception(
+            "Please enter a supported action -> renameBaseTables, createMatViews, getCacherScript, restructAirbyteDbt"
+        )
+
+
+# Function to grab Airbyte Generated dbt and move into CTA structured folder (✨Progify it✨)
+def restructure_airbyte_dbt(airbyte_workspace_path, dbt_cta_path):
+    # Mapping of Airbyte folders -> CTA folders
+    airbyte_cta_dir_mapping = {
+        "airbyte_ctes": "0_ctes",
+        "airbyte_tables": "1_cta_full_refresh",
+        "airbyte_incremental": "1_cta_incremental",
+    }
+
+    # Go through and move all files from Airbyte export into 'dbt-cta' directories
+    for airbyte_dir, cta_dir in airbyte_cta_dir_mapping.items():
+        # Get all sql files that should be moved for current mapping
+        files_to_move = glob(
+            f"{airbyte_workspace_path}/**/{airbyte_dir}/**/*.sql", recursive=True
+        )
+
+        if files_to_move:  # Only run if there were any files found
+            # Make sure target path is created
+            dbt_cta_path = (
+                dbt_cta_path if dbt_cta_path[-1] != "/" else dbt_cta_path[:-1]
+            )
+            target_path = f"{dbt_cta_path}/models/{cta_dir}"
+            os.makedirs(target_path, exist_ok=True)
+
+            # Move files to target path
+            for file_path in files_to_move:
+                file_name = file_path.split("/")[-1]
+                moved_file = move(src=file_path, dst=f"{target_path}/{file_name}")
+                print(f"Moved {file_path} -> {moved_file}")
+
+    # Now lets move over the sources.yml file
+    sources_file = glob(
+        f"{airbyte_workspace_path}/**/models/**/sources.yml", recursive=True
+    )
+    if sources_file:
+        move_source_file = move(
+            src=sources_file[0], dst=f"{dbt_cta_path}/models/sources.yml"
+        )
+        print(f"Moved {sources_file[0]} -> {dbt_cta_path}/models/sources.yml")
+    else:
+        print(f"Couldnt find the sources file in {airbyte_workspace_path}")
+
+    # Cool, files are moved so now lets modify the file content
+    modify_dbt_models(f"{dbt_cta_path}/models/0_ctes", "cte")
+    modify_full_refresh_models(f"{dbt_cta_path}/models/1_cta_full_refresh")
+    modify_dbt_models(f"{dbt_cta_path}/models/1_cta_incremental", "incremental")
+
+
+# Function to modify ctes and incremental models. For these its just gonna be
+# swapping out the config section, and changing source to cta.
+def modify_dbt_models(path_to_models, model_type):
+    # Default Config Section thats gonna be used to replace Airbyte generated config
+    config = (
+        "{{ config(\n"
+        '    cluster_by = "_airbyte_emitted_at",\n'
+        '    partition_by = {"field": "_airbyte_emitted_at", "data_type": "timestamp", "granularity": "day"},\n'
+        '    unique_key = "_airbyte_ab_id"\n'
+        ") }}\n"
+    )
+    stop_skip_pattern = "-- SQL model" if model_type == "cte" else "-- Final base SQL"
```

Comment:
yeah lol so there will very possibly come a time when Airbyte changes something and this strat won't work anymore 🫠🫠🫠 Probably this strat makes sense for now, but wonder if A) there are any other ways we can logically achieve the same thing, and B) we should probably flag this in a comment or the readme or have some error handling or something (like, maybe raise an exception if neither of these patterns are present?)

Reply:
yeahhh this definitely a problem. The other thing that came up while testing this is that the macros change... i dont really know what the best approach for fixing that other than running the dbt and seeing what fails. I honestly think the best solution is to make our own version of the default normalizer or i guess just live with the fact that this will just need constant upkeep. But yeah i can add some kind of a check here

Reply:
> I honestly think the best solution is to make our own version of the default normalizer

This would be the dream! Fisher and I derped around once in the Airbyte repo to try and find the code they use to run default normalizations. We were more or less unsuccessful... I think whatever process they use is buried very deep within the codebase, and maybe even in like Java or something. I wonder how much of an undertaking this would be? It would make our processes much more resilient to use Airbyte ONLY up until the point where it delivers raw JSON, and then we could have code that flattens the JSON and creates the dbt models we crave... It seems doable, at least in theory 🧐 This would be a heftychomnk project though so I'd like us to discuss with Fisher. I'll draft a ticket tho 😤